src/app/projects/projects.service.ts

```diff
@@ -1,16 +1,130 @@
-import { Injectable } from '@angular/core';
+import { Injectable, OnDestroy } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import { map } from 'rxjs/operators';
-import { Observable } from 'rxjs';
+import { map, switchMap, tap } from 'rxjs/operators';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
 import { environment } from '../../environments/environment';
-import { Project } from './project';
+import { PaginatedProjects, Project } from './project';
+
+interface Filters {
+  organ: string;
+  technology: string;
+  location: string;
+  searchVal: string;
+  recentFirst: boolean;
+}
 
 @Injectable()
-export class ProjectsService {
-  constructor(private http: HttpClient) {}
-  URL = `${environment.ingestApiUrl}${environment.catalogueEndpoint}`;
+export class ProjectsService implements OnDestroy {
+  private URL = `${environment.ingestApiUrl}${environment.catalogueEndpoint}`;
+
+  private projectsPerPage = 20;
```

Comment:
I think that is too many. 20 projects per page means a lot of scrolling in the page. I would decrease. We could add in the future a drop down to select page size.
```suggestion
  private projectsPerPage = 10;
```